js/lib/Loader.js

```diff
@@ -88,8 +88,16 @@ Loader.prototype._loadFileAlongIncludePath = function(includePath, pathname) {
 
 Loader.prototype.loadFiles = function(paths, sync, params, callback, root) {
     root = root || (params && params.base);
-    var includePath = root ? [root].concat(this.includePath) : this.includePath;
+    var includePath = this.includePath;
 
+    if (root) {
+        if (ilib._load.multiPaths){
+            includePath.splice(1, 0, root);
+        } else  {
```

Comment:
Before this change, we have:

1. root
2. includePaths (standard paths from the loader)

After this change, we have:

1. includePath[0] (the first one in the list, which is presumably the first multiPath)
2. root
3. includePath[1-X] (the second and subsequent multiPaths up to X)
4. includePath[X+1 - N] (the standard includePaths)

I think this is not what we want.

How about this:

this.includePath -> always the standard list that the loader defines
root -> always the root
this.multiPath -> any paths added/removed by the client using addPath() and removePath(). this.multiPath is initialized to the empty array [] in the constructor;

Then, the loadFiles should set:

```
var includePath = [].concat(this.multiPath); // makes a copy
if (root) includePath.push(root);
includePath.concat(this.includePath);

```

That way, we end up with:

1. All the multiPaths first
2. root
3. standard include paths from the loader

What do you think?

Reply:
Codes would be more organized and neatly. I'll update it following your suggestion.